### ISSUE TITLE
Improve unit tests for chipper.

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/chipper#1012]
Based on https://github.com/phetsims/chipper/issues/1011 ...

Looking at unit tests for chipper, coverage is inadequate.  All top-level task (those shown in grunt --help) should have a niminimum of 1 test.  

Labeled for developer meeting to decide who to assign.

**Comment by pixelzoom** [comment]
Units tests are also apparently not run in CT.  They should be.

**Comment by zepumph** [comment]
@samreid and I ran into this today when we created `CHIPPER/phetioCompareAPIsTests`. Note that command line qunit is already set up to run everything in `chipper/test`. That seems like the way to handle this. 

Likely with qunit as an added dependency to chipper, and a new CT test type that can run an arbitrary command line argument.

**Comment by samreid** [comment]
From today's dev meeting:

Aqua already runs builds on CT, so that covers some ground.
JO: Asks about the cost-benefit analysis for adding tests.  Would we `grunt rc` on bumper as part of this?
SR: That sounds like overkill to me.
CM: Maybe we would choose the things not already exercised by CT, that developers use rarely (such as create-sim), and add tests for them.

(x) So we would like to volunteer someone to add a test for create-sim, and to evaluate which other tasks may need tests.  This would run the create-sim task, and make sure it doesn't fail. Then the output can run in puppeteer, and have a built version.

JB: Can QA do this manually so we don't have to make time to automate this?
CM: They are already very busy.

(x) CM: How about running the unit tests that we have in CT?

(x) CM: We can delete milagro and migrate.
SR: I can do those parts.

I'll create side issues for everything with an (x) and we can meet next time to delegate the sub-issues.

**Comment by samreid** [comment]
I think I'll repurpose this issue for running chipper/perennial tests on CT, and I'll create a side issue for testing create-sim specifically.

**Comment by jessegreenberg** [comment]
4/29/21

Today we discussed that it would be nice to have a general way to run tests tests. We should create another issue related to security and running commands. @jonathanolson will create that issue and create a general way to create tests for scripts for usage in issues like the ones listed above.

For shell vs grunt vs node tasks - there is a general consensus that creating a node task for this makes most sense so we don't add more grunt tasks.

**Comment by marlitas** [comment]
@samreid and I looked through this issue, and it seems it might need dev team discussion again to decide what the priority, need, and subgroup for this should be.

**Comment by marlitas** [comment]
This issue is connected and should be assigned to the same subgroup:
https://github.com/phetsims/perennial/issues/225

**Comment by marlitas** [comment]
https://github.com/phetsims/chipper/issues/1037 is connected and should be assigned to the same subgroup.

**Comment by marlitas** [comment]
JO: Chipper/pernnial are in a state where they are not actively being checked. 
KP: Once the new CT server is added in (which is priority), this would be good to add as continued improvement to CT
SR: All four chipper tests are currently passing.
KP: Let's connect this to the server epic.
SR: It would not take many resources to get the test coverage to be more robust. This issue also dovetails with https://github.com/phetsims/chipper/issues/1272. This would go under the umbrella of improving chipper/perennial.
